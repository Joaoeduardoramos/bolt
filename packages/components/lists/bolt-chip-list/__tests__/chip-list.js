import {
  render,
  renderString,
  stop as stopTwigRenderer,
} from '@bolt/twig-renderer';

async function renderTwig(template, data) {
  return await render(template, data, true);
}

async function renderTwigString(template, data) {
  return await renderString(template, data, true);
}

describe('<bolt-chip-list> Component', async () => {
  afterAll(async () => {
    await stopTwigRenderer();
  });

  test('basic usage without links', async () => {
    const results = await renderTwig('@bolt-components-chip-list/chip-list.twig', {
      items: [
        {
          text: 'Chip text 1',
        },
        {
          text: 'Chip text 2',
        },
        {
          text: 'Chip text 3',
        },
        {
          text: 'Chip text 4',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('basic usage with links', async () => {
    const results = await renderTwig('@bolt-components-chip-list/chip-list.twig', {
      items: [
        {
          text: 'Chip link 1',
          url: '#!',
        },
        {
          text: 'Chip link 2',
          url: '#!',
        },
        {
          text: 'Chip link 3',
          url: '#!',
        },
        {
          text: 'Chip link 4',
          url: '#!',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  test('basic usage with mixed behavior', async () => {
    const results = await renderTwig('@bolt-components-chip-list/chip-list.twig', {
      items: [
        {
          text: 'Chip link 1',
          url: '#!',
        },
        {
          text: 'Chip text 1',
        },
        {
          text: 'Chip link 3',
          url: '#!',
        },
        {
          text: 'Chip text 2',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
