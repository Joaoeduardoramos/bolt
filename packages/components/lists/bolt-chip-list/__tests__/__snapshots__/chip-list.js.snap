// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-chip-list> Component basic usage with links 1`] = `

<bolt-list tag="ul"
           display="inline"
           spacing="xsmall"
           separator="none"
           align="start"
           valign="center"
>
  <replace-with-grandchildren>
    <ul class="c-bolt-list c-bolt-list--display-inline c-bolt-list--spacing-xsmall c-bolt-list--align-start c-bolt-list--valign-center">
      <bolt-list-item>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start">
            <bolt-chip bolt-component>
              <a href="#!"
                 class="c-bolt-chip"
              >
                <span class="c-bolt-chip__item-text">
                  Chip link 1
                </span>
              </a>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
      <bolt-list-item>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start">
            <bolt-chip bolt-component>
              <a href="#!"
                 class="c-bolt-chip"
              >
                <span class="c-bolt-chip__item-text">
                  Chip link 2
                </span>
              </a>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
      <bolt-list-item>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start">
            <bolt-chip bolt-component>
              <a href="#!"
                 class="c-bolt-chip"
              >
                <span class="c-bolt-chip__item-text">
                  Chip link 3
                </span>
              </a>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
      <bolt-list-item last>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start c-bolt-list-item--last-item">
            <bolt-chip bolt-component>
              <a href="#!"
                 class="c-bolt-chip"
              >
                <span class="c-bolt-chip__item-text">
                  Chip link 4
                </span>
              </a>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
    </ul>
  </replace-with-grandchildren>
</bolt-list>

`;

exports[`<bolt-chip-list> Component basic usage with mixed behavior 1`] = `

<bolt-list tag="ul"
           display="inline"
           spacing="xsmall"
           separator="none"
           align="start"
           valign="center"
>
  <replace-with-grandchildren>
    <ul class="c-bolt-list c-bolt-list--display-inline c-bolt-list--spacing-xsmall c-bolt-list--align-start c-bolt-list--valign-center">
      <bolt-list-item>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start">
            <bolt-chip bolt-component>
              <a href="#!"
                 class="c-bolt-chip"
              >
                <span class="c-bolt-chip__item-text">
                  Chip link 1
                </span>
              </a>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
      <bolt-list-item>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start">
            <bolt-chip bolt-component>
              <span class="c-bolt-chip">
                <span class="c-bolt-chip__item-text">
                  Chip text 1
                </span>
              </span>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
      <bolt-list-item>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start">
            <bolt-chip bolt-component>
              <a href="#!"
                 class="c-bolt-chip"
              >
                <span class="c-bolt-chip__item-text">
                  Chip link 3
                </span>
              </a>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
      <bolt-list-item last>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start c-bolt-list-item--last-item">
            <bolt-chip bolt-component>
              <span class="c-bolt-chip">
                <span class="c-bolt-chip__item-text">
                  Chip text 2
                </span>
              </span>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
    </ul>
  </replace-with-grandchildren>
</bolt-list>

`;

exports[`<bolt-chip-list> Component basic usage without links 1`] = `

<bolt-list tag="ul"
           display="inline"
           spacing="xsmall"
           separator="none"
           align="start"
           valign="center"
>
  <replace-with-grandchildren>
    <ul class="c-bolt-list c-bolt-list--display-inline c-bolt-list--spacing-xsmall c-bolt-list--align-start c-bolt-list--valign-center">
      <bolt-list-item>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start">
            <bolt-chip bolt-component>
              <span class="c-bolt-chip">
                <span class="c-bolt-chip__item-text">
                  Chip text 1
                </span>
              </span>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
      <bolt-list-item>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start">
            <bolt-chip bolt-component>
              <span class="c-bolt-chip">
                <span class="c-bolt-chip__item-text">
                  Chip text 2
                </span>
              </span>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
      <bolt-list-item>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start">
            <bolt-chip bolt-component>
              <span class="c-bolt-chip">
                <span class="c-bolt-chip__item-text">
                  Chip text 3
                </span>
              </span>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
      <bolt-list-item last>
        <replace-with-grandchildren>
          <li class="c-bolt-list-item c-bolt-list-item--display-inline c-bolt-list-item--spacing-xsmall c-bolt-list-item--align-start c-bolt-list-item--last-item">
            <bolt-chip bolt-component>
              <span class="c-bolt-chip">
                <span class="c-bolt-chip__item-text">
                  Chip text 4
                </span>
              </span>
            </bolt-chip>
          </li>
        </replace-with-grandchildren>
      </bolt-list-item>
    </ul>
  </replace-with-grandchildren>
</bolt-list>

`;
